libs/payments/webhooks/src/lib/subscriptionHandler.service.ts

```diff
@@ -34,11 +36,21 @@ export class SubscriptionEventsService {
     const price = subscription.items.data[0].price;
 
     let paymentProvider: PaymentProvidersType | undefined;
-    let voluntaryCancellation: boolean | undefined;
+    let determinedCancellation: boolean | undefined;
+    let optedOut: boolean | undefined;
     try {
       const customer = await this.customerManager.retrieve(
         subscription.customer
       );
+      if (customer.metadata['userid']) {
+        const accountResp = await this.accountManager.getAccounts([
+          customer.metadata['userid'],
+        ]);
+        optedOut =
+          accountResp && accountResp.length > 0
+            ? accountResp[0].metricsOptOutAt !== null
+            : false;
+      }
```

Comment:
First, see my previous comment regarding if opt out is necessary.

If it is necessary, then instead of duplicating logic, could the `userid` be passed along with the other event data to the emitter service, and use private method `EmitterService.retrieveOptOut` in `EmitterService.handleSubscriptionEnded`?